sim-lib/src/lib.rs

```diff
@@ -733,65 +751,101 @@ async fn consume_events(
     mut receiver: Receiver<SimulationEvent>,
     sender: Sender<SimulationOutput>,
     shutdown: Trigger,
+    listener: Listener,
 ) {
     log::debug!("Started consumer for {}.", node.lock().await.get_info());
 
-    while let Some(event) = receiver.recv().await {
-        match event {
-            SimulationEvent::SendPayment(dest, amt_msat) => {
-                let mut node = node.lock().await;
-
-                let mut payment = Payment {
-                    source: node.get_info().pubkey,
-                    hash: None,
-                    amount_msat: amt_msat,
-                    destination: dest.pubkey,
-                    dispatch_time: SystemTime::now(),
-                };
-
-                let outcome = match node.send_payment(dest.pubkey, amt_msat).await {
-                    Ok(payment_hash) => {
-                        log::debug!(
-                            "Send payment: {} -> {}: ({}).",
-                            node.get_info(),
-                            dest,
-                            hex::encode(payment_hash.0)
-                        );
-                        // We need to track the payment outcome using the payment hash that we have received.
-                        payment.hash = Some(payment_hash);
-                        SimulationOutput::SendPaymentSuccess(payment)
-                    }
-                    Err(e) => {
-                        log::error!(
-                            "Error while sending payment {} -> {}.",
-                            node.get_info(),
-                            dest
-                        );
-
-                        match e {
-                            LightningError::PermanentError(s) => {
-                                log::error!("Simulation terminated with error: {s}.");
-                                shutdown.trigger();
-                                break;
+    loop {
+        select! {
+            biased;
+            _ = listener.clone() => {
+                let node = node.lock().await;
+                let node_info = node.get_info();
+                log::debug!("Simulation event consumer for {node_info} received shutdown signal");
```

Comment:
nit: inline `log::debug!("Simulation event consumer for {} received shutdown signal", node.lock().await.get_info());`